### PR TITLE
Add an example with StrMap to reduce

### PR DESCRIPTION
**Pull request by kapooostin**
In the wake of [this comment](https://gitter.im/sanctuary-js/sanctuary?at=5e2595ce20d0f078a6ad6587) on Gitter

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1901,6 +1901,9 @@
   //.
   //. > reduce (concat, '', Cons ('foo', Cons ('bar', Cons ('baz', Nil))))
   //. 'foobarbaz'
+  //.
+  //. > reduce (concat, '', {foo: 'ary', bar: 'Sa', baz: 'nctu'})
+  //. 'Sanctuary'
```
Comment (davidchambers) [inline index.js]: I find this example more complex than necessary. I'll see whether I have permission to push an update to this branch.

File: index.js [hunk 2]
```diff
@@ -1901,6 +1901,9 @@
   //.
   //. > reduce (concat, '', Cons ('foo', Cons ('bar', Cons ('baz', Nil))))
   //. 'foobarbaz'
+  //.
+  //. > reduce (concat, '', {foo: 'x', bar: 'y', baz: 'z'})
+  //. 'yzx'
```
Comment (davidchambers) [inline index.js]: What do you think of this example, @kapooostin?

Thank you very much for submitting this pull request, by the way. :clap:

Comment (kapooostin) [reply index.js]: Great, thanks you! Now it is illustrative enough without unnecessary complications.

**Comment by kapooostin** [comment]
David, i accepted the changes you suggested. Have I forgotten to press some button here in order to proceed with this PR?

**Comment by davidchambers** [comment]
Thanks again for this contribution, @kapooostin. I am sorry for being so slow to merge it!

**Pull request merged**